plugin/trino-kafka/src/main/java/io/trino/plugin/kafka/schema/confluent/AvroConfluentContentSchemaReader.java

```diff
@@ -47,7 +51,11 @@ public AvroConfluentContentSchemaReader(SchemaRegistryClient schemaRegistryClien
         }
         checkState(dataSchemaLocation.isEmpty(), "Unexpected parameter: dataSchemaLocation");
         try {
-            return Optional.of(schemaRegistryClient.getLatestSchemaMetadata(subject.get()).getSchema());
+            SchemaMetadata schemaMetadata = schemaRegistryClient.getLatestSchemaMetadata(subject.get());
+            ParsedSchema schema = schemaRegistryClient.getSchemaById(schemaMetadata.getId());
```

Comment:
How does it resolve schema reference?

Reply:
Schemas are resolved internally by the `SchemaProvider` implementation. In case of `AvroSchema#getRawSchema` - It gives a fully resolved Schema (which is based on `SchemaReferences` that is passed)